src/main/java/org/testng/TestNG.java

```diff
@@ -1090,8 +1061,17 @@ else if (m_masterfileName == null) {
     }
   }
 
-  private void p(String string) {
-    System.out.println("[TestNG] " + string);
+  /**
+   * Run the test suites.
+   * <p>
+   * This method can be overridden by subclass. <br/>
+   * For example, DistributedTestNG to run in master/slave mode according to commandline args. 
+   * </p>
+   * @return
+   * @since 6.9.11 when moving distributed/remote classes out into separate project
+   */
+  protected List<ISuite> runSuites() {
```

Comment:
introduce new override-able method `runSuites` to abstract how to run the suites, this is helpful for DistributedTestNG to override this method to run master/slave mode according to command line args, see also: https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/distributed/src/main/java/org/testng/distributed/DistributedTestNG.java#L54